projects/kit/src/lib/utils/time/parse-time-string.ts

```diff
@@ -3,17 +3,38 @@ import type {MaskitoTimeSegments} from '../../types';
 /**
  * @param timeString can be with/without fixed characters
  */
-export function parseTimeString(timeString: string): Partial<MaskitoTimeSegments> {
+export function parseTimeString(
+    timeString: string,
+    timeMode: string,
+): Partial<MaskitoTimeSegments> {
     const onlyDigits = timeString.replaceAll(/\D+/g, '');
 
+    const sliceIndexes = createSliceIndexes(timeMode);
+
     const timeSegments = {
-        hours: onlyDigits.slice(0, 2),
-        minutes: onlyDigits.slice(2, 4),
-        seconds: onlyDigits.slice(4, 6),
-        milliseconds: onlyDigits.slice(6, 9),
+        hours: onlyDigits.slice(...sliceIndexes.hours),
+        minutes: onlyDigits.slice(...sliceIndexes.minutes),
+        seconds: onlyDigits.slice(...sliceIndexes.seconds),
+        milliseconds: onlyDigits.slice(...sliceIndexes.milliseconds),
     };
 
     return Object.fromEntries(
         Object.entries(timeSegments).filter(([_, value]) => Boolean(value)),
     );
 }
+
+function createSliceIndexes(timeMode: string): MaskitoTimeSegments<[number, number]> {
+    const offset =
+        Number(timeMode.startsWith('MM')) * 2 +
+        Number(timeMode.startsWith('SS')) * 4 +
+        Number(timeMode.startsWith('MSS')) * 6;
+
+    const changeSelection = (index: number): number => Math.max(index - offset, 0);
+
+    return {
+        hours: [changeSelection(0), changeSelection(2)],
+        minutes: [changeSelection(2), changeSelection(4)],
+        seconds: [changeSelection(4), changeSelection(6)],
+        milliseconds: [changeSelection(6), changeSelection(9)],
```

Comment:
What do you think about this solution?

```ts
import type {MaskitoTimeMode, MaskitoTimeSegments} from '../../types';

const SEGMENT_FULL_NAME: Record<string, keyof MaskitoTimeSegments> = {
    HH: 'hours',
    MM: 'minutes',
    SS: 'seconds',
    MSS: 'milliseconds',
};

/**
 * @param timeString can be with/without fixed characters
 */
export function parseTimeString(
    timeString: string,
    timeMode: MaskitoTimeMode,
): Partial<MaskitoTimeSegments> {
    const onlyDigits = timeString.replaceAll(/\D+/g, '');

    let offset = 0;

    return Object.fromEntries(
        timeMode.split(/\W/).map(segmentAbbr => {
            const segmentValue = onlyDigits.slice(offset, offset + segmentAbbr.length);

            offset += segmentAbbr.length;

            return [SEGMENT_FULL_NAME[segmentAbbr], segmentValue];
        }),
    );
}

```

Reply:
I think it's better. It uses easier logic